cloudevents/pkg/reconciler/events/cache/cache.go

```diff
@@ -0,0 +1,26 @@
+package cache
+
+import (
+	"context"
+	"errors"
+	cloudevents "github.com/cloudevents/sdk-go/v2"
+)
+
+// AddEventSentToCache adds the particular object to cache marking it as sent
+func AddEventSentToCache(ctx context.Context, event *cloudevents.Event) error {
+	cacheClient := Get(ctx)
+	if cacheClient == nil {
+		return errors.New("cache client is nil")
+	}
+	cacheClient.Add(event.String(), nil)
```

Comment:
Thanks for adding this. 

I think we'll have to use something different for the key though, as duplicate events might not be exactly the same.
Even with the exact same pipelinerun, timestamps will be different. 

We need to define a function that extracts the part of an event that we consider enough to define two events as the same, and use that as a key.  Probably a combination for event type, resource type and name should be enough to send each event type once for each resource.

In future we might want to send more events, or refresher kind of events, when something changes in the pipeline run but the condition stays the same. We could use the same kind of conditions that are used in the cloud events controller in pipeline today.

Reply:
Fixed